### ISSUE TITLE
Toggling required does not effect form validity

### ISSUE THREAD

**Issue by FritzHerbers** [ngx-formly/ngx-formly#1493]
```
[x] bug report
```

**Current behavior**
Toggling required does not effect form validity.

**Minimal reproduction of the problem with instructions**

https://stackblitz.com/edit/angular-nr5wln?file=src%2Fapp%2Fapp.component.html

See reproduction steps in the example.

**Side note**
The examples use the bootstrap class "button-default", which has been renamed to "button-secondary".

**Comment by aitboudad** [comment]
see https://github.com/ngx-formly/ngx-formly/issues/1362#issuecomment-451643594

**Comment by FritzHerbers** [comment]
@aitboudad 
The linked comment is about checkboxes and to validate a checked state.

My case is with a blank text input field. The validity of the form does not change when `required` is changed.

**Comment by aitboudad** [comment]
sorry, sometimes I read the title only :smile:,

 sure if it worth to support your use-case :thinking:, we support that when `required` in `expressionProperties` https://stackblitz.com/edit/angular-nr5wln-65mcuc but not let me think about it.

**Comment by aitboudad** [comment]
added a note in our roadmap #1365

**Comment by FritzHerbers** [comment]
private note:
A formControl is not removed when for a container the hide flag is set.
hideExpression has to be used on the container, so that the formControl is removed.

https://stackblitz.com/edit/angular-nr5wln-8mqbvd?file=src/app/app.component.ts

**Comment by aitboudad** [comment]
not true, the above example works as expected for me.

**Comment by FritzHerbers** [comment]
it's a working example / no issue, just wanted to note that setting the hide flag, won't remove the formControl.

**Comment by aitboudad** [comment]
fixed in `v5.5.3`.

**Comment by FritzHerbers** [comment]
Thanks for the fix.

Personal note:
Reduces the way to have to use something like
```
expressionProperties: {
   "templateOptions.required": "formState.required"
}
```

Only a delete required does not change validity; use toggle required.
